### PR TITLE
Prevent Shrinking of Certain Small Borg Chassis w/ Small Refactor

### PR DESCRIPTION
**Pull request by KasparoVy**
## About The Pull Request
This PR exists b/c #5211 got closed, an ultimatum PR (#6210) got created and I'd really like to keep the ability to shrink borgs (who aren't already dangerously small)!

The OG PR 5211 missed the very model that people like to screenshot for this issue but this PR catches it.

This also refactors the `dogborg` and `has_snowflake_deadsprite` variables into this list of features.

## Why It's Good For The Game
You can't shrink borgs that are small off the get-go so they're still clickable in combat, and cleans up code - making altborgs a bit more flexible if we choose to further leverage the features list in the future while also keeping the borg shrinker module's function for those who enjoy it with most other chassis types.

## Changelog
:cl:
rscdel: The borg shinker module will no longer work on certain compact chassis models, providing a feedback message. This includes the peacekeeper-ball, vroomba, eyeborg, mining drone and some others.
tweak: SkyRat Altborg code's been refactored - indicating that a model is small, is a dogborg or has unique wreck sprites is now done in a features list.
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/altborgs/code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 1]
```diff
@@ -539,25 +520,23 @@
 		if("Omni")
 			cyborg_base_icon = "omoikane"
 			cyborg_icon_override = 'modular_skyrat/modules/altborgs/icons/mob/robots_pk.dmi'
+			model_features = list(R_TRAIT_SMALL) //No tennis-ball sized cyborgs.
```
Comment (KasparoVy) [inline modular_skyrat/modules/altborgs/code/modules/mob/living/silicon/robot/robot_modules.dm:524]: This' the one people like to screenshot.

File: modular_skyrat/modules/altborgs/code/modules/mob/living/silicon/robot/robot_model.dm [hunk 2]
```diff
@@ -1,10 +1,9 @@
 /obj/item/robot_model
 	var/icon/cyborg_icon_override
 	var/sleeper_overlay
-	var/has_snowflake_deadsprite
 	var/cyborg_pixel_offset
 	var/model_select_alternate_icon
-	var/dogborg = FALSE //Is this model a wider borg?
+	var/list/model_features //Traits unique to this model, i.e. having a unique dead sprite, being wide or being small enough to reject shrinker modules. Leverages defines in code\__DEFINES\~skyrat_defines\robot_defines.dm
```
Comment (tf-4) [inline modular_skyrat/modules/altborgs/code/modules/mob/living/silicon/robot/robot_model.dm]: Could autodoc this

Comment (KasparoVy) [reply modular_skyrat/modules/altborgs/code/modules/mob/living/silicon/robot/robot_model.dm]: Is that the triple slash format?
EDIT: https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md - going to follow this, thanks for the suggestion

**Comment by KasparoVy** [comment]
Another commit coming with a fix, a bit more refactoring and some documentation

**Comment by KasparoVy** [comment]
Tested on local and everything that previously checked the `.dogborg` variable still seems to work using the `is_dogborg()` check
Also this PR's diff is now perfectly balanced

**Review by Funce — APPROVED** [review]
This is great! Solid, and I like it.

**Comment by KasparoVy** [comment]
Sweet, cheers! I'm glad we could figure something out

EDIT: 1 last force-push from a nitpick I had, just wanted to document in the borg shapeshifter, define it as a list and initialize them properly

**Pull request merged**